idflakies-maven-plugin/src/main/java/edu/illinois/cs/dt/tools/plugin/IncDetectorMojo.java

```diff
@@ -0,0 +1,635 @@
+                for (File file : files) {
+                    String fileName = file.toString().substring(file.toString().lastIndexOf(File.separator) + 1);
+                    if (fileName.endsWith(".clz")) {
+                        String transitiveClosureKey = fileName.substring(0, fileName.indexOf(".clz"));
+                        BufferedReader in = Files.newBufferedReader(Paths.get(file.toString()), StandardCharsets.UTF_8);
+                        String str;
+                        Set<String> transitiveClosureValue = new HashSet<>();
+                        while ((str = in.readLine()) != null) {
+                            if (!str.contains(".class") || !str.startsWith("file:")) {
+                                continue;
+                            }
+                            int sepIndex = str.lastIndexOf("_");
+                            String urlExternalForm = str.substring(0, sepIndex);
+                            URL url = new URL(urlExternalForm);
+                            String transitiveClosureValueArrayItem = getClassNameFromClassFile(url.getPath());
+                            transitiveClosureValue.add(transitiveClosureValueArrayItem);
+                        }
+                        transitiveClosure.put(transitiveClosureKey, transitiveClosureValue);
+                    }
+                }
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+            // the dependency map from classes to their dependencies (test classes)
+            reverseTransitiveClosure = getReverseClosure(transitiveClosure);
+        } else {
+            try {
+                List<String> zlcLines = Files.readAllLines(startsDependenciesPath, Charset.defaultCharset());
+                String firstLine = zlcLines.get(0);
+                String space = " ";
+
+                // check whether the first line is for *
+                if (firstLine.startsWith(STAR_FILE) || firstLine.equals("PLAIN_TEXT")) {
+                    zlcLines.remove(0);
+                }
+
+                for (String line : zlcLines) {
+                    String[] parts = line.split(space);
+                    String stringURL = parts[0];
+                    Set<String> tests = parts.length == 3 ? fromCSV(parts[2]) : new HashSet<String>();
+                    URL url = new URL(stringURL);
+                    String reverseTransitiveClosureKey = getClassNameFromClassFile(url.getPath());
+                    reverseTransitiveClosure.put(reverseTransitiveClosureKey, tests);
+                }
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+            // the dependency map from test classes to their dependencies (classes)
+            transitiveClosure = getReverseClosure(reverseTransitiveClosure);
+        }
+        return transitiveClosure;
+    }
+
+    private void checkSelectAll() throws IOException, MojoExecutionException {
```

Comment:
Is the logic used here for checking the exact same as what STARTS would rely on?